File: MS.NET/Applications/Desktop/WpfMvvmTest/BindingApp/VisitorModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BindingApp
{
    public class Visitor
    {
        public string Name { get; set; }

        public int Frequency { get; set; }

        public DateTime Recent { get; set; }

        public Visitor()
        {
            Frequency = 1;
            Recent = DateTime.Now;
        }

        public void Revisit()
        {
            Frequency += 1;
            Recent = DateTime.Now;
        }
    }

    public class VisitorModel
    {
        private Document<Visitor> dataStore;

        public VisitorModel()
        {
            dataStore = Document.Open<Visitor>("visitors.xml");
        }

        public Visitor[] ReadVisitors()
        {
            return dataStore.ToArray();
        }

        public void WriteVisitor(string id)
        {
            Visitor visitor = dataStore.Find(entry => entry.Name == id);

            if (visitor == null)
                dataStore.Add(new Visitor { Name = id });
            else
                visitor.Revisit();

            dataStore.Save();
        }
    }
}

File: Java/Applications/Desktop/SwingMVCTest/VisitorModel.java
package app;

public class VisitorModel extends javax.swing.table.AbstractTableModel{

		private Document<Visitor> store;

		public VisitorModel(){
			store = Document.open(Visitor.class, "visitors.xml");
		}

		@Override
		public int getColumnCount(){
			return 3;
		}

		@Override
		public String getColumnName(int i){
			String[] names = {"Name", "Frequency", "Recent"};
			return names[i];
		}

		@Override
		public int getRowCount(){
			return store.size();
		}

		@Override
		public Object getValueAt(int i, int j){
			Visitor entry = store.get(i);
			Object[] values = {entry.name, entry.frequency, entry.recent};
			return values[j];
		}

		public void register(String id){
			Visitor visitor = store.find(entry -> entry.name.equals(id));
			if(visitor == null)
				store.add(new Visitor(id));
			else
				visitor.revisit();
			store.save();
			fireTableDataChanged();
		}
}


